Tests/scripts/collect_tests/collect_tests.py

```diff
@@ -141,8 +144,18 @@ def __init__(
             logger.info(f'collected {test=}, {reason} ({reason_description}, {version_range=})')
 
         if pack:
-            self.packs = {pack}
-            logger.info(f'collected {pack=}, {reason} ({reason_description}, {version_range=})')
+            if only_to_upload == only_to_install:
```

Comment:
Yes, but also `not only_to_upload and not only_to_install` (both are True or both are False)

Reply:
a bit confusing, according to your docstring:
```
        :param only_to_install: whether to collect the pack only to install it without upload to the bucket.
        :param only_to_upload: whether to collect the pack only to upload it to the bucket without install.
```
it can never be both True.

Reply:
Sorry, my bad, it can't be both True, `only_to_install=True` will be for test dependencies which will never be only_to_upload.
Added a verification for this.

Reply:
do we still need this condition?